app/helpers/uploads_helper.rb

```diff
@@ -0,0 +1,35 @@
+require 'fileutils'
+require 'audit'
+require 'open3'
+
+module UploadsHelper
+  class Postprocessor
+    def self.process(f)
+      if self.respond_to?(File.extname(f)[1..-1])
```

Comment:
(Presumably your security hole is if I named a file `foobar.to_s` or something like that where a predefined method by that name exists?  Hence the dictionary request, with an object that has absolutely no other methods on it besides the ones we give it.)